### ISSUE TITLE
Mention when a new feature was introduced in the docs

### ISSUE THREAD

**Issue by OmarBazaraa** [ClickHouse/clickhouse-docs#2125]
ClickHouse gets new features constantly and continuously, and the new features, settings, configurations, ..etc. get documented regularly  on https://clickhouse.com/docs. It will be so nice and useful if the docs mention when a new feature was introduced or changed.

For example, for lightweight deletes, it will be nice if the docs mentions when the feature:
1. Was first introduced as an experimental feature
2. Became production ready and available by default in the server

Another example would be for settings and configurations. If the default value of a server setting changed from `x` to `y` on version `v`, then it will be useful to mention in the docs that:
1. The default value is `x`.
2. And starting from version `v`, the default value became `y`.

Basically mentioning the history of features and default values.

And thanks so much for the great product, ClickHouse!

**Comment by tbragin** [comment]
cc @melvynator FYI

**Comment by johnnymatthews** [comment]
I've given this the `X-Large` size flag since it'll be a huge amount of work to go through all the existing docs/settings/functions/features and add this _added in version XYZ_ status. I'm not saying we _shouldn't do this_ -- have this status would be super valuable. I'm just setting expectations that this isn't an easy nor quick fix.

**Comment by OmarBazaraa** [comment]
Yeah I understand it'd be a huge effort. Starting with the new features/changes would still be beneficial tho.
And thanks a lot!

**Comment by Andersson007** [comment]
Opened an issue with an alternative solution https://github.com/ClickHouse/clickhouse-docs/issues/2492

**Comment by gingerwizard** [comment]
@Andersson007 comment

> Hello,
> 
> The community is developing an [Ansible collection for ClickHouse](https://github.com/ansible-collections/community.clickhouse) and often struggles with the docs having no versions.
> 
> There was a similar issue reported in https://github.com/ClickHouse/clickhouse-docs/issues/2125 but i would suggest not mentioning the version (it'd also work and better than nothing) but creating version-specific docs, in other words, be able to choose a version of ClickHouse similar like in PostgreSQL
> 
> ![Screenshot from 2024-07-16 09-56-31](https://github.com/user-attachments/assets/d1549cae-43a7-41ed-b6d7-5b410bca66da)
> 
> It could be based on building docs from corresponding branches matching clickhouse versions.
> 
> Would make user life way easier.

**Comment by Blargian** [comment]
Not a complete solution to this problem yet, but we have introduced a 'Version history' dropdown to Session and MergeTree settings pages: https://clickhouse.com/docs/operations/settings/settings. With this you will be able to see when the most commonly used settings were introduced and in which versions their default values changed.
